#!/usr/bin/env python
PACKAGE = "detection_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_dist", double_t, 0, "Maximum distance from measurement to filter", 0.1, 0, 2)
gen.add("max_cov", double_t, 0, "Maximum covariance before filter is killed", 10, 0, 10)
gen.add("max_pub_cov", double_t, 0, "Maximum covariance to publish filter", 0.1, 0, 1)
gen.add("max_pub_vel", double_t, 0, "Maximum velocity to publish filter", 0.02, 0, 1)

gen.add("process_noise_cov", double_t, 0, "Process Noise Covariance", 0.05, 0, 1)
gen.add("measurement_noise_cov", double_t, 0, "Measurement Noise Covariance", 0.5, 0, 1)
gen.add("error_cov_post", double_t, 0, "Error Covariance Post", 0.5, 0, 1)
gen.add("period", double_t, 0, "Measurement Period", 0.3, 0, 2)

gen.add("PDgO", double_t, 0, "Probability of detection given object", 0.9, 0, 1)
gen.add("PDgo", double_t, 0, "Probability of detection given no object", 0.1, 0, 1)
gen.add("PO_init", double_t, 0, "Initial probabiliy of object", 0.1, 0, 1)
gen.add("pub_certainty_thresh", double_t, 0, "Certainty Threshold for publication", 0.9, 0, 1)
gen.add("min_certainty", double_t, 0, "Minimum certainty to keep filter alive", 0.01, 0, 1)

gen.add("odometry_tick_dist", double_t, 0, "Distance traveled at which out of view filters are predicted", 5, 0, 20)
gen.add("exclusion_zone_range", double_t, 0, "Distance traveled to keep exclusion zones", 30.0, 0, 1000)

gen.add("max_colormodel_distance", double_t, 0,
    "Maximum color model distance for same filter", 0.3, 0, 1.0)

gen.add("positive_exclusion_radius", double_t, 0, "Radius of exclusion zone for collected object", 10, 0, 50)
gen.add("negative_exclusion_radius", double_t, 0, "Radius of exclusion zone for false positive object", 1.5, 0, 10)

gen.add("clear_filters",bool_t, 0, "Clear all filters", False)

exit(gen.generate(PACKAGE, "detection_filter", "kalman_filter_params"))
